lib/renderer/renderer-webpack.js

```diff
@@ -0,0 +1,524 @@
+                                        stats.stats.forEach(stat => {
+                                            if (stat.hasErrors()) {
+                                                console.error(stat.compilation.errors);
+                                            }
+                                        });
+                                    } else if (stats.hasErrors()) {
+                                        console.error(stats.compilation.errors);
+                                    }
+                                }
+                                console.debug(`Cached ${filepath}`);
+                                // we use a counter because it's an async function and can go out of sync
+                                count++;
+                                if (count === filepaths.length) {
+                                    console.debug(`--------\nFinished Precaching`);
+                                }
+                            }
+                        });
+                } catch (error) {
+                    console.error(error);
+                }
+            } catch (error) {
+                console.error(`Error Precacheing ${filepath} \n\n ${error}`);
+            }
+        }
+    }
+    /**
+     * @param {string} filePath
+     * @param {string} vueFile
+     * @returns {Promise<WebpackConfigType>}
+     */
+    async BuildConfig(filePath, vueFile) {
+        return new Promise((resolve, reject) => {
+            const parsedBundlePath = path.parse(filePath);
+            const app = `import Vue from "vue";
+            import App from "${filePath}";
+
+            export function createApp (data) {
+                const app = new Vue({
+                    data,
+                    render: h => h(App)
+                })
+                return { app }
+            }`;
+
+            const server = `import { createApp } from "./app"
+            export default context => {
+                return new Promise((resolve, reject) => {
+                    const { app } = createApp(context);
+
+                    resolve(app);
+                });
+            };`;
+
+            const client = `import { createApp } from "./app"
+            const store = window.__INITIAL_STATE__;
+            const { app } = createApp(store ? store : {});
+            app.$mount("#app")`;
+
+            const parsedFilePath = filePath.replace(this.trueRootPath, "").replace(".vue", "");
+            const configFolder = path.join(this.trueRootPath, ".expressvue", parsedFilePath);
+            try {
+                fs.statSync(configFolder).isDirectory();
+            } catch (error) {
+                mkdirp.sync(configFolder);
+            }
+
+            const appPath = path.join(configFolder, "app.js");
+            const serverPath = path.join(configFolder, "entry-server.js");
+            const clientPath = path.join(configFolder, "entry-client.js");
+
+            fs.writeFileSync(appPath, app, "utf-8");
```

Comment:
Why write file sync and not async?
```suggestion
            await Promise.all([fs.writeFile(appPath, app, "utf-8"),fs.writeFile(serverPath, server, "utf-8"), fs.writeFile(clientPath, client, "utf-8")])
```

Reply:
`fs.writeFile` takes a callback, it's not a promise.

the `fs.promises` is still new and experimental, and only in node 10